api/tests/functional-tests/backend/core/test_llm_clients.py

```diff
@@ -0,0 +1,302 @@
+    monkeypatch.setattr(
+        "valor_api.backend.core.llm_clients.LLMClient.__call__",
+        _return_invalid_response,
+    )
+
+    with pytest.raises(ValueError):
+        client.answer_relevance("some query", "some text")
+
+    # patch __call__ with a valid response
+    monkeypatch.setattr(
+        "valor_api.backend.core.llm_clients.LLMClient.__call__",
+        _return_valid_coherence_response,
+    )
+
+    assert 5 == client.coherence("some text")
+
+    # patch __call__ with an invalid response
+    monkeypatch.setattr(
+        "valor_api.backend.core.llm_clients.LLMClient.__call__",
+        _return_invalid_response,
+    )
+
+    with pytest.raises(ValueError):
+        client.coherence("some text")
+
+
+def test_WrappedOpenAIClient():
+    def create_bad_request(model, messages, seed) -> ChatCompletion:
+        raise ValueError
+
+    def create_mock_chat_completion_with_bad_length(
+        model, messages, seed
+    ) -> ChatCompletion:
+        return ChatCompletion(
+            id="foo",
+            model="gpt-3.5-turbo",
+            object="chat.completion",
+            choices=[
+                Choice(
+                    finish_reason="length",
+                    index=0,
+                    message=ChatCompletionMessage(
+                        content="some response",
+                        role="assistant",
+                    ),
+                )
+            ],
+            created=int(datetime.datetime.now().timestamp()),
+        )
+
+    def create_mock_chat_completion_with_content_filter(
+        model, messages, seed
+    ) -> ChatCompletion:
+        return ChatCompletion(
+            id="foo",
+            model="gpt-3.5-turbo",
+            object="chat.completion",
+            choices=[
+                Choice(
+                    finish_reason="content_filter",
+                    index=0,
+                    message=ChatCompletionMessage(
+                        content="some response",
+                        role="assistant",
+                    ),
+                )
+            ],
+            created=int(datetime.datetime.now().timestamp()),
+        )
+
+    def create_mock_chat_completion(model, messages, seed) -> ChatCompletion:
+        return ChatCompletion(
+            id="foo",
+            model="gpt-3.5-turbo",
+            object="chat.completion",
+            choices=[
+                Choice(
+                    finish_reason="stop",
+                    index=0,
+                    message=ChatCompletionMessage(
+                        content="some response",
+                        role="assistant",
+                    ),
+                )
+            ],
+            created=int(datetime.datetime.now().timestamp()),
+        )
+
+    client = WrappedOpenAIClient(api_key=None, model_name="model_name")
+
+    fake_message = [{"key": "value"}]
+
+    # with pytest.raises(openai.OpenAIError):
+    #     client.connect()
```

Comment:
Why should this client.connect() fail? Should it throw an error if OPENAI_API_KEY is set? It wouldn't fail for me locally.

Reply:
that's right. it fails for me (and on the GitHub runner) since we don't have the OPENAI_API_KEY set. you could remove this test if you'd like; agree that this isn't necessary